syntax = "proto3";
package Proto;

import "Request.proto";
import "Reply.proto";
import "Exchange.proto";

enum StrategyType
{
  Quoter = 0;
  Hitter = 1;
  Dimer = 2;
  DummyQuoter = 3;
}

//enum StrategyStatusEnum
//{
//  Play = 0;
//  Stop = 1;
//  Running = 2;
//  Pause = 3;
//  Error = 4;
//  Lost = 5; /// heartbeat lost
//}

enum StrategyOperation
{
  Start = 0;
  Pause = 1;
  Stop = 2;
}

enum StrategyStatus
{
  Running = 0;
  Started = 2;
  Stopped = 3;
  Paused = 4;
  Error = 5;
  Lost = 6; /// heartbeat lost
}

message StrategyOperate
{
  string name = 1;
  string underlying = 2;
  StrategyOperation operate = 3;
}

message StrategyOperateReq
{
  RequestType type = 1;
  Exchange exchange = 2;
  string user = 3;
  repeated StrategyOperate operates = 4;
}

//message StrategyStatusRep
//{
//  string error = 1;
//}

message StrategySwitch
{
  StrategyType strategy = 1;
  string option = 2;
  bool is_bid = 3;
  bool is_ask = 4;
  bool is_qr_cover = 5;
}

message StrategySwitchReq
{
  RequestType type = 1;
  Exchange exchange = 2;
  string user = 3;
  StrategyType strategy = 4;
  repeated StrategySwitch switches = 5;
}

message StrategySwitchRep
{
  repeated StrategySwitch switches = 1;
  Reply result = 2;
}

message StrategyStatistic
{
  string name = 1;
  StrategyType type = 2;
  Exchange exchange = 3;
  string underlying = 4;
  StrategyStatus status = 5;
  double delta = 6;
  int32 orders = 7;
  int32 trades = 8;
}
